dgisim/src/dices.py

```diff
@@ -1,29 +1,62 @@
 from __future__ import annotations
-
 import random
+from collections import Counter
+from functools import lru_cache
+from typing import Any, Optional, Iterator, Iterable
+from typing_extensions import Self, override, TYPE_CHECKING
 from collections import defaultdict
-from typing import Optional, Iterator, Iterable, Union, List, Set, Tuple, Dict
+from functools import cache
+from heapq import heappop, heapify
+
+from typing_extensions import TYPE_CHECKING, Self
 
-from typing_extensions import Self
+if TYPE_CHECKING:
+    from .state.game_state import GameState
+    from .state.player_state import PlayerState
```

Comment:
I normally put the `if TYPE_CHECKING` part after the import from other packages of this project.

Reply:
done